src/Aspire.Hosting/ApplicationModel/BeforeResourceStartedEvent.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Aspire.Hosting.Eventing;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// This event is raised by orchestrators before they have started a new resource.
+/// </summary>
+/// <param name="resource">The resource that is being created.</param>
+/// <param name="services">The <see cref="IServiceProvider"/> for the app host.</param>
+/// <remarks>
+/// Resources that are created by orchestrators may not yet be ready to handle requests.
+/// </remarks>
+[Experimental("ASPIREEVENTING001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+public class BeforeResourceStartedEvent(IResource resource, IServiceProvider services) : IDistributedApplicationResourceEvent
```

Comment:
I think this name still violates the naming guidelines. 

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-events

✔️ DO give events names with a concept of before and after, using the present and past tenses.

For example, a close event that is raised before a window is closed would be called Closing, and one that is raised after the window is closed would be called Closed.

❌ DO NOT use "Before" or "After" prefixes or postfixes to indicate pre- and post-events. Use present and past tenses as just described.

```suggestion
public class ResourceStartingEvent(IResource resource, IServiceProvider services) : IDistributedApplicationResourceEvent
```

Reply:
Agree these names aren't 100% right. But we will be doing a follow up.